### DISCUSSION TITLE
Necessity of SymbolizeConstantLoopBounds pass

### DISCUSSION THREAD

**Discussion by JakeSilverman** [seahorn/seahorn#561] [Q&A]
I am writing a termination analysis that uses seahorn as the frontend to convert c code into smt2 format. I see that one of the optimization passes, "SymbolizeConstantLoopBounds", creates an infinite loop (SymbolizeConstantLoopBounds.cc line 195). Clearly this affects the results of a termination analysis. I also see the seahorn intentionally omits adding a flag to turn this pass off (seahorn.cpp line 362). I am curious as to the reason for this omission - if I manually turn off this pass am I going to break some other seahorn functionality?

**Comment by caballa** [comment]
That pass is used only to help Spacer, the CHC solver used by SeaHorn. If you disable it you won't break any SeaHorn functionality.
